libs/zsdl/src/sdl2.zig

```diff
@@ -394,6 +399,25 @@ pub const Texture = opaque {
         }
     }
     extern fn SDL_QueryTexture(texture: *Texture, format: ?*u32, access: ?*c_int, w: ?*c_int, h: ?*c_int) c_int;
+
+    pub fn lock(texture: ?*Texture, rect: ?*Rect) !PixelData {
+        var pixels: *anyopaque = undefined;
+        var pitch: i32 = undefined;
+        if (SDL_LockTexture(texture, rect, &pixels, &pitch) != 0) {
+            return makeError();
+        }
+
+        return PixelData{
+            .pixels = @ptrCast(pixels),
+            .pitch = pitch,
+        };
+    }
+    extern fn SDL_LockTexture(texture: ?*Texture, rect: ?*Rect, pixels: ?**anyopaque, pitch: ?*c_int) c_int;
```

Comment:
I think the `texture` fn args here should not be optional.

Reply:
Fixed this up.